Doc/library/asyncio-stream.rst

```diff
@@ -206,12 +206,20 @@ StreamReader
 
    .. coroutinemethod:: read(n=-1)
 
-      Read up to *n* bytes.  If *n* is not provided, or set to ``-1``,
-      read until EOF and return all read bytes.
+      Read up to *n* bytes.
 
+      If *n* is not provided, or set to ``-1``,
+      read until EOF. After EOF is received, return all read bytes.
```

Comment:
I'm actually unsure whether the details about the character code are correct. I suppose the standard use case of the `StreamReader` is to read from a TCP socket. It seems to me that this maps to the system's `recv` function (though, I didn't look through all details). The man page on a Debian Linux says:

> When a stream socket peer has performed an orderly shutdown, the return value will be 0 (the traditional "end-of-file" return).

Similarly, the man page on my macOS says:

> If no messages are available to be received and the peer has performed an orderly shutdown, the value 0 is returned.

This sounds to me like there is no actual character code read. (I suppose a TCP FIN packet will be received, which is handled by the network stack and causes read just to return 0.)

I suppose for a different kind of stream (file instead of socket?), a character code might be used to indicate EOF, but at that point it would become a detail of the underlying stream and be no longer a general aspect of the `StreamReader`.

But I might also be misunderstanding things? I'm not that familiar with these low-level details.

(The other changes look good to me btw.)

Reply:
“EOF” refers to the concept of ending the stream, not to a character. ^D and others are not passed to the application (except maybe in raw tty mode) and should not be mentioned here.